src/main/java/com/example/demo/model/User.java

```diff
@@ -26,4 +33,41 @@ public class User {
     @Column(nullable = false)
     private String lastName;
     private String shippingAddress;
+    @ManyToMany
+    @JoinTable(
+            name = "user_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return roles;
+    }
+
+    @Override
+    public String getUsername() {
+        return email;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
```

Comment:
```suggestion
        return !isDeleted;
```
Let’s  implement soft-deleted

Reply:
if i haven't endpoint wich delete user i still need soft delete

Reply:
yes, you are setting this up for an account deactivation feature that may be implemented in the future